File: solutions/java/CodeWars/src/main/java/com/kata/kyu6/stringSubpatternRecognitionI/test/StringSubpatternRecognitionITest.java
package com.kata.kyu6.stringSubpatternRecognitionI.test;

import com.kata.kyu6.stringSubpatternRecognitionI.main.StringSubpatternRecognitionI;
import org.junit.Test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

public class StringSubpatternRecognitionITest {
    @Test
    public void basicTests() {
        assertFalse(StringSubpatternRecognitionI.hasSubpattern("a"));
        assertTrue(StringSubpatternRecognitionI.hasSubpattern("aaaa"));
        assertFalse(StringSubpatternRecognitionI.hasSubpattern("abcd"));
        assertTrue(StringSubpatternRecognitionI.hasSubpattern("abababab"));
        assertFalse(StringSubpatternRecognitionI.hasSubpattern("ababababa"));
        assertTrue(StringSubpatternRecognitionI.hasSubpattern("123a123a123a"));
        assertFalse(StringSubpatternRecognitionI.hasSubpattern("123A123a123a"));
        assertTrue(StringSubpatternRecognitionI.hasSubpattern("abbaabbaabba"));
        assertFalse(StringSubpatternRecognitionI.hasSubpattern("abbabbabba"));
        assertFalse(StringSubpatternRecognitionI.hasSubpattern("abcdabcabcd"));
    }
}

File: solutions/csharp/Kata/Kata/6-kyu/String_subpattern_recognition_I/test/StringSubpatternRecognitionITest.cs
using Kata._6_kyu.String_subpattern_recognition_I.src;
using NUnit.Framework;

namespace Kata._6_kyu.String_subpattern_recognition_I.test;

public sealed class StringSubpatternRecognitionITest
{
    [Test]
    public void FixedTest()
    {
        Assert.That(StringSubpatternRecognitionI.HasSubpattern("a"), Is.EqualTo(false));
        Assert.That(StringSubpatternRecognitionI.HasSubpattern("aaaa"), Is.EqualTo(true));
        Assert.That(StringSubpatternRecognitionI.HasSubpattern("abcd"), Is.EqualTo(false));
        Assert.That(StringSubpatternRecognitionI.HasSubpattern("abababab"), Is.EqualTo(true));
    }
}